modules/stochastic_tools/src/samplers/CartesianProductSampler.C

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CartesianProductSampler.h"
+#include "Distribution.h"
+
+registerMooseObject("StochasticToolsApp", CartesianProductSampler);
+
+defineLegacyParams(CartesianProductSampler);
+
+InputParameters
+CartesianProductSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Provides complete cartesian product for the supplied variables.");
+  params.addRequiredParam<std::vector<Real>>(
+      "items", "A list of items, each item should include a min, step size, and number of steps.");
```

Comment:
The name `item` doesn't really mean anything, and the description would leave a user pretty lost as to what this is. Maybe call it something like `intervals`, `ranges_and_steps`, `ranges_and_divisions`? This is something that should be described in a little more depth in the .md file.

Reply:
I think I will leave "items", but will improve the docs.

Reply:
Whaaaat???  I think 'items' is pretty much the least descriptive name you could come up with for a parameter!  I think I'm going to start naming all of my parameters 'items1', 'items2', 'items3', etc.

Reply:
Or maybe 'stuff1', 'stuff2', 'stuff3'...

Reply:
Wow, I struck a nerve. The problem is that I can't come up with anything that isn't also misleading, your suggestions also don't hit the meaning for me as well.

What about creating multiple parameters:
```
starts = `10 20`
steps = `1  2`
counts = `10 20`
```

Reply:
OK, maybe I was being a little dramatic!

I think something like what you suggest would be an ideal way to make this more clear.  Do you mean 'steps' rather than 'stops', though?

Reply:
Maybe `linear_spacing`? Since it's similar to the `linspace` function in python and Matlab. This would also give the opportunity for defining `log_spacing` as a separate option.